Tools/wasm/emscripten/__main__.py

```diff
@@ -159,6 +159,26 @@ def make_build_python(context, working_dir):
     print(f"🎉 {binary} {version}")
 
 
+@subdir(HOST_DIR, clean_ok=True)
+def make_emscripten_libffi(context, working_dir):
+    shutil.rmtree(HOST_DIR/"libffi", ignore_errors=True)
+
+    call([
+        "git",
+        "clone",
+        "https://github.com/libffi/libffi",
+        "--depth=1"
+    ],
+    quiet=context.quiet,
+    )
+    call(
+        [EMSCRIPTEN_DIR / "make_libffi.sh"],
+        env=updated_env({"PREFIX": HOST_DIR}),
+        cwd=HOST_DIR/"libffi",
+        quiet=context.quiet,
```

Comment:
This results in the libffi build and install existing *inside* the CPython checkout; that makes it difficult to do a clean CPython build without also requiring a full libffi rebuild. Would it make sense to keep the libffi checkout, build and install separate? 

The Android build script (which has a similar "install the dependencies" step) uses `/cross-build/aarch64-linux-android/prefix` as an install prefix for the dependencies, and `/cross-build/aarch64-linux-android/build` as the CPython checkout - it might make sense to follow the same pattern here.